<?xml version="1.0" encoding="UTF-8"?>
  
<launch>
	
        <arg name="leader" default="leader"/>
        <arg name="ego" default="ego"/>

        <arg name="SPEED_SCALE" default="1.0"/>
        <arg name="HEADWAY_SCALE" default="1.0"/>	
        <arg name="ego_veltopic" default="/$(arg ego)/vel"/>
        <arg name="relative_veltopic" default="/$(arg ego)/rel_vel"/>
        <arg name="headway_topic" default="/$(arg ego)/lead_dist"/>
        <arg name="use_accel_predict" default="true"/>
        <arg name="use_lead_vel" default="false"/>
        <arg name="use_odom" default="false"/>
        <arg name="ego_odom_topic" default="/$(arg ego)/setvel"/>
        <arg name="leader_odom_topic" default="/$(arg leader)/setvel"/>
        <arg name='hoffman' default='false'/>

        <arg name="T" default="0.6"/>
        <arg name="th1" default="0.4"/>
        <arg name="th2" default="0.6"/>
        <arg name="th3" default="0.8"/>
        <arg name="w1" default="1.0"/>
        <arg name="w3" default="2.5"/>
        <arg name="w2" default="3.0"/>

        <arg name='controllers' default='09_05_rll_ttc6p5scale1p8_maxacc0p75/super_resolution.onnx'/>

        <param name="/$(arg ego)/model" type="string" value="$(find onnx2ros)/../controllers/$(arg controllers)"/>
        <param name="/$(arg ego)/mode" type="string" value="prompt"/>
        <param name="/$(arg ego)/SPEED_SCALE" value="$(arg SPEED_SCALE)"/>
        <param name="/$(arg ego)/HEADWAY_SCALE" value="$(arg HEADWAY_SCALE)"/>
        <param name="/$(arg ego)/ego_vel_topic" value="$(arg ego_veltopic)"/>
        <param name="/$(arg ego)/relative_vel_topic" value="$(arg relative_veltopic)"/>
        <param name="/$(arg ego)/headway_topic" value="$(arg headway_topic)"/>
        <param name="/$(arg ego)/use_lead_vel" value="$(arg use_lead_vel)"/>
        <param name="/$(arg ego)/use_accel_predict" value="$(arg use_accel_predict)"/>
        <param name="/$(arg ego)/use_odom" value="$(arg use_odom)"/>
        <param name="/$(arg ego)/ego_odom_topic" value="$(arg ego_odom_topic)"/>
        <param name="/$(arg ego)/leader_odom_topic" value="$(arg leader_odom_topic)"/>
        <param name="/$(arg ego)/use_setpoint" value="false"/>
        <param name="/$(arg ego)/T" value="$(arg T)"/>

        
	<group ns="$(arg ego)">
        <node pkg="onnx2ros" type="prompt_mode" name="$(arg ego)_rlcontroller" output="screen">
                <remap from="v_des" to="/$(arg ego)/cmd_vel" unless="$(arg hoffman)"/>
                <remap from="v_des" to="/$(arg ego)/cmd_control_vel" if="$(arg hoffman)"/>
                <remap from="leader_vel" to="/$(arg leader)/vel"/>
        </node>
        <node pkg="hoffmansubsystem" type="hoffmansubsystem_node" name="hoffmansubsystem_$(arg leader)" output="screen" if="$(arg hoffman)"/>
	</group>

	
</launch>	
